src/main/java/duck/task/EventTask.java

```diff
@@ -0,0 +1,91 @@
+package duck.task;
+
+import static duck.Parser.OUTPUT_DATE_FORMAT;
+
+import java.time.LocalDate;
+
+import duck.DuckException;
+
+/**
+ * Represents a event task.
+ */
+public class EventTask extends Task {
+    private LocalDate start;
+    private LocalDate end;
+
+    /**
+     * Creates a new EventTask.
+     * 
+     * @param name   Name of the task.
+     * @param isDone Status of the task.
+     * @param start Start of the task.
+     * @param end End of the task.
+     */
+    public EventTask(String name, boolean isDone, LocalDate start, LocalDate end) {
+        super(name, isDone);
+        this.start = start;
+        this.end = end;
+    }
+
+    private String formatStart() {
+        return start.format(OUTPUT_DATE_FORMAT);
+    }
+
+    private String formatEnd() {
+        return end.format(OUTPUT_DATE_FORMAT);
+    }
+
+    @Override
+    public String stringify() {
+        String startString = formatStart();
+        String endString = formatEnd();
+        return "E" + super.stringify()
+                + startString.length() + "/" + startString
+                + endString.length() + "/" + endString;
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString()
+                + " (from: " + formatStart() + " to " + formatEnd() + ")";
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof EventTask) {
+            EventTask other = (EventTask) obj;
+            return super.equals(other) && start.equals(other.start) && end.equals(other.end);
+        }
```

Comment:
Perhaps you can split the boolean checking up in order to reduce complicated expressions